test/integration/hotrestart_test.sh

```diff
@@ -66,7 +66,7 @@ JSON_TEST_ARRAY+=("${HOT_RESTART_JSON_REUSE_PORT}")
 
 # Enable this test to work with --runs_per_test
 if [[ -z "${TEST_RANDOM_SEED}" ]]; then
-  BASE_ID=1
+  BASE_ID=$$
```

Comment:
shall we always use the shell pid? it works even in runs_per_test right?

Reply:
ping?

Reply:
Sorry, got distracted by work. I'll double check, but yes I think it's safe to just use the PID always.

Reply:
So, it appears that $$ when is always 16. I guess that's probably true for coverage as well.I think TEST_SHARD_INDEX should be the fallback there. I'll up date this to do that.

Reply:
And even better is that TEST_RANDOM_SEED is repeated for each test, so if you do `--runs_per_test=5` with multiple tests, they still collide.